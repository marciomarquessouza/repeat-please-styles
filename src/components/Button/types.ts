import { TextStyle, ViewStyle } from 'react-native';

export interface IBtuttonProps {
	children: string;
	style?: ViewStyle;
	textStyle?: TextStyle;
	onPress?: () => void;
	isLoading?: boolean;
}
