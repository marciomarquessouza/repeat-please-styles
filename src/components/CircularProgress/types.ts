export interface ICircularProgressProps {
	lateralMargin?: number;
	barWidth?: number;
	duration: number;
	progress: number;
	startAnimation: boolean;
	chartText: string;
}

export interface ICircularProgressState {
	showCircleUp: boolean;
	showMonkeyHead: boolean;
	showTextChart: boolean;
}
