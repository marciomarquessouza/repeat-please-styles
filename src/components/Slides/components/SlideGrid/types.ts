import { Animated, PanResponderInstance } from 'react-native';

export interface ISlide {
	header: JSX.Element | null;
	footer: JSX.Element | null;
}

export interface ISlideGridProps {
	slides: ISlide[];
	duration: number;
	nextScreen: () => void;
}

export interface ISlideGridState {
	position: Animated.ValueXY;
	panResponder: PanResponderInstance;
	index: number;
	lastPosition: number;
}
