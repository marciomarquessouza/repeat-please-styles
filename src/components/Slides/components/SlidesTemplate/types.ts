export interface ISlideProps {
	title: string;
	text: string;
}
