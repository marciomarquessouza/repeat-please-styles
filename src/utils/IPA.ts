export const IPA = {
	ʌ: {
		symbol: 'ʌ',
		type: 'vowel',
		examples: ['cUp', 'lUck'],
	},
	'ɑ:': {
		symbol: 'ɑ:',
		type: 'vowel',
		examples: ['Arm', 'fAther'],
	},
	æ: {
		symbol: 'æ',
		type: 'vowel',
		examples: ['cAt', 'blAck'],
	},
	e: {
		symbol: 'e',
		type: 'vowel',
		examples: ['mEt', 'bEd'],
	},
	ə: {
		symbol: 'ə',
		type: 'vowel',
		examples: ['Away', 'cinemA'],
	},
	'ɜ:': {
		symbol: 'ɜ:',
		type: 'vowel',
		examples: ['tURn', 'lEARn'],
	},
	ɪ: {
		symbol: 'ɪ',
		type: 'vowel',
		examples: ['hIt', 'sITTING'],
	},
	'i:': {
		symbol: 'i:',
		type: 'vowel',
		examples: ['SEE', 'HEAT'],
	},
	ɒ: {
		symbol: 'ɒ',
		type: 'vowel',
		examples: ['HOT', 'ROCK'],
	},
	'ɔ:': {
		symbol: 'ɔ:',
		type: 'vowel',
		examples: ['CALL', 'FOUR'],
	},
	ʊ: {
		symbol: 'ʊ',
		type: 'vowel',
		examples: ['PUT', 'COULD'],
	},
	'u:': {
		symbol: 'u:',
		type: 'vowel',
		examples: ['BLUE', 'FOOD'],
	},
	aɪ: {
		symbol: 'aɪ',
		type: 'vowel',
		examples: ['FIVE', 'EYE'],
	},
	aʊ: {
		symbol: 'aʊ',
		type: 'vowel',
		examples: ['NOW', 'OUT'],
	},
	eɪ: {
		symbol: 'eɪ',
		type: 'vowel',
		examples: ['SAY', 'EIGHT'],
	},
	oʊ: {
		symbol: 'oʊ',
		type: 'vowel',
		examples: ['GO', 'HOME'],
	},
	ɔɪ: {
		symbol: 'ɔɪ',
		type: 'vowel',
		examples: ['BOY', 'JOIN'],
	},
	eə: {
		symbol: 'eə',
		type: 'vowel',
		examples: ['WHERE', 'AIR'],
	},
	ɪə: {
		symbol: 'ɪə',
		type: 'vowel',
		examples: ['NEAR', 'HERE'],
	},
	ʊə: {
		symbol: 'ʊə',
		type: 'vowel',
		examples: ['PURE', 'TOURIST'],
	},
};
