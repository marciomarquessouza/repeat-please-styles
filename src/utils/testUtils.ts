import { ShallowWrapper } from 'enzyme';

export const findByDataTest = (
	wrapper: ShallowWrapper,
	dataTest: string,
): ShallowWrapper => {
	return wrapper.find(`[data-test="${dataTest}"]`);
};
