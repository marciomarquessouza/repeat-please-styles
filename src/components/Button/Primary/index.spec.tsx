import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ButtonPrimary } from './index';
import { findByDataTest } from '../../../utils/testUtils';
import { IBtuttonProps } from '../types';

const defaultProps: IBtuttonProps = {
	children: 'Primary',
};

const setup = (props: IBtuttonProps): ShallowWrapper => {
	return shallow(<ButtonPrimary {...props}>{props.children}</ButtonPrimary>);
};

describe('Button Primary Default Tests', () => {
	it('should render the button components properly', () => {
		const wrapper = setup(defaultProps);
		const buttonPrimary = findByDataTest(wrapper, 'buttonPrimary');
		expect(buttonPrimary).toHaveLength(1);
		expect(buttonPrimary).toMatchSnapshot();
	});

	it('should render the loading properly', () => {
		const wrapper = setup({ ...defaultProps, isLoading: true });
		const loading = findByDataTest(wrapper, 'loading');
		expect(loading).toHaveLength(1);
		expect(loading).toMatchSnapshot();
	});
});
