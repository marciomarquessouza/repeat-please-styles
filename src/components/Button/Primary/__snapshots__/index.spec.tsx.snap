// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Primary Default Tests should render the button components properly 1`] = `ShallowWrapper {}`;

exports[`Button Primary Default Tests should render the loading properly 1`] = `ShallowWrapper {}`;
